File: app/src/main/java/closer/vlllage/com/closer/api/models/GroupActionResult.java
package closer.vlllage.com.closer.api.models;

import closer.vlllage.com.closer.store.models.GroupAction;

public class GroupActionResult extends ModelResult {
    public String group;
    public String name;
    public String intent;
    public String photo;

    public static GroupAction from(GroupActionResult groupActionResult) {
        GroupAction groupAction = new GroupAction();
        groupAction.setId(groupActionResult.id);
        updateFrom(groupAction, groupActionResult);
        return groupAction;
    }

    public static GroupAction updateFrom(GroupAction groupAction, GroupActionResult groupActionResult) {
        groupAction.setName(groupActionResult.name);
        groupAction.setIntent(groupActionResult.intent);
        groupAction.setGroup(groupActionResult.group);
        groupAction.setPhoto(groupActionResult.photo);
        return groupAction;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/api/models/GroupActionResult.kt
package closer.vlllage.com.closer.api.models

import closer.vlllage.com.closer.store.models.GroupAction
import java.util.*

class GroupActionResult : ModelResult() {
    var group: String? = null
    var name: String? = null
    var intent: String? = null
    var about: String? = null
    var photo: String? = null
    var flow: String? = null
    var used: Date? = null

    companion object {

        fun from(groupActionResult: GroupActionResult): GroupAction {
            val groupAction = GroupAction()
            groupAction.id = groupActionResult.id
            updateFrom(groupAction, groupActionResult)
            return groupAction
        }

        fun updateFrom(groupAction: GroupAction, groupActionResult: GroupActionResult): GroupAction {
            groupAction.name = groupActionResult.name
            groupAction.intent = groupActionResult.intent
            groupAction.about = groupActionResult.about
            groupAction.group = groupActionResult.group
            groupAction.photo = groupActionResult.photo
            groupAction.flow = groupActionResult.flow
            groupAction.used = groupActionResult.used
            return groupAction
        }
    }
}
